src/main/java/com/awakenedredstone/autowhitelist/discord/BotHelper.java

```diff
@@ -74,6 +79,12 @@ public static void sendTempSimpleMessage(MessageChannel channel, Text message, i
         messageAction.queue(m -> m.delete().queueAfter(seconds, TimeUnit.SECONDS));
     }
 
+    public static List<Role> getRolesForMember(Member member) {
+        List<Role> roles = member.getRoles();
```

Comment:
```suggestion
        List<Role> roles = new ArrayList(member.getRoles());
```
MemberImpl#getRoles returns `Collections.unmodifiableList(roleList)`
I was trying to avoid doing this to not risk mem leak but I think I'm being worried for nothing

Other than that, everything looks great

Reply:
That's what I would bet, given https://stackoverflow.com/questions/53767722/converting-immutable-to-mutable-list-java-is-there-any-alternative and the dig into it shows a literal transmutation of the underlying iterable into a new memory object

Reply:
(in my rebasing I somehow lost the arraylist change, so added that back)

Reply:
> That's what I would bet, given https://stackoverflow.com/questions/53767722/converting-immutable-to-mutable-list-java-is-there-any-alternative and the dig into it shows a literal transmutation of the underlying iterable into a new memory object

Yeah, that is why I had a worry on that, but the GC should take care of that, else Java would blow up